### ISSUE TITLE
[Feature request] Icons next to folder/file names

### ISSUE THREAD

**Issue by sum01** [NicolaiSoeborg/filemanager-plugin#11]
Stuff like [vim-devicons](https://github.com/ryanoasis/vim-devicons) or [seti-icons](https://atom.io/packages/seti-icons) would be cool to have.

**Comment by tommyshem** [comment]
Yes, it would be nice to have. I will look into it when I get time.

**Comment by tommyshem** [comment]
Looking into this and it looks like you would have to install a font with the devicons in the font set, like nerdfonts on your system to get this to work. As micro editor can only use one font from the terminal settings. If you have any different ideas please let me know.

**Comment by sum01** [comment]
Either bundle the fonts & get the path to them using the `configDir` variable, or make an option that's `false` by default, which when enabled adds the icons to paths in `scanDir`

**Comment by runtimefailure** [comment]
the way to get icons to work is not hard and i share the code of it i don't know how long it will take to be added by the author but it can be done locally by editing the plugin source code that is on your device.

this is how it currently looks like
<img width="310" height="169" alt="Image" src="https://github.com/user-attachments/assets/696902a2-fd7d-4f32-8d7c-8b09b677521f" />

its using the nerd font for the icons and this is the code that needs to be replaced bellow line 138

	local devicons = {
	extensions = {
		styl = ' ',
		sass = ' ',
		scss = ' ',
		htm = ' ',
		html = ' ',
		slim = ' ',
		ejs = ' ',
		css = ' ',
		less = ' ',
		md = ' ',
		mdx = ' ',
		markdown = ' ',
		rmd = ' ',
		json = ' ',
		js = ' ',
		mjs = ' ',
		jsx = ' ',
		rb = ' ',
		php = ' ',
		py = ' ',
		pyc = ' ',
		pyo = ' ',
		pyd = ' ',
		coffee = ' ',
		mustache = ' ',
		hbs = ' ',
		conf = ' ',
		ini = ' ',
		yml = ' ',
		yaml = ' ',
		toml = ' ',
		bat = ' ',
		jpg = ' ',
		jpeg = ' ',
		bmp = ' ',
		png = ' ',
		webp = ' ',
		gif = ' ',
		ico = ' ',
		twig = ' ',
		cpp = ' ',
		['c++'] = ' ',
		cxx = ' ',
		cc = ' ',
		cp = ' ',
		c = ' ',
		cs = ' ',
		h = ' ',
		hh = ' ',
		hpp = ' ',
		hxx = ' ',
		hs = ' ',
		lhs = ' ',
		lua = ' ',
		java = ' ',
		sh = ' ',
		fish = ' ',
		bash = ' ',
		zsh = ' ',
		ksh = ' ',
		csh = ' ',
		awk = ' ',
		ps1 = ' ',
		ml = 'λ ',
		mli = 'λ ',
		diff = ' ',
		db = ' ',
		sql = ' ',
		dump = ' ',
		clj = ' ',
		cljc = ' ',
		cljs = ' ',
		edn = ' ',
		scala = ' ',
		go = ' ',
		dart = ' ',
		xul = ' ',
		sln = ' ',
		suo = ' ',
		pl = ' ',
		pm = ' ',
		t = ' ',
		rss = ' ',
		['f#'] = ' ',
		fsscript = ' ',
		fsx = ' ',
		fs = ' ',
		fsi = ' ',
		rs = ' ',
		rlib = ' ',
		d = ' ',
		erl = ' ',
		hrl = ' ',
		ex = ' ',
		exs = ' ',
		eex = ' ',
		leex = ' ',
		vim = ' ',
		ai = ' ',
		psd = ' ',
		psb = ' ',
		ts = ' ',
		tsx = ' ',
		jl = ' ',
		pp = ' ',
		vue = '﵂ ',
		elm = ' ',
		swift = ' ',
		xcplayground = ' ',
	},
	exact_matches = {
		['exact-match-case-sensitive-1.txt'] = '1',
		['exact-match-case-sensitive-2'] = '2',
		['gruntfile.coffee'] = ' ',
		['gruntfile.js'] = ' ',
		['gruntfile.ls'] = ' ',
		['gulpfile.coffee'] = ' ',
		['gulpfile.js'] = ' ',
		['gulpfile.ls'] = ' ',
		['mix.lock'] = ' ',
		dropbox = '  ',
		['.ds_store'] = ' ',
		['.gitconfig'] = ' ',
		['.gitignore'] = ' ',
		['.gitlab-ci.yml'] = ' ',
		['.bashrc'] = ' ',
		['.zshrc'] = ' ',
		['.vimrc'] = ' ',
		['.gvimrc'] = ' ',
		_vimrc = ' ',
		_gvimrc = ' ',
		['.bashprofile'] = ' ',
		['favicon.ico'] = ' ',
		license = ' ',
		node_modules = ' ',
		['react.jsx'] = ' ',
		procfile = ' ',
		dockerfile = ' ',
		['docker-compose.yml'] = ' ',
		makefile = ' ',
		['cmakelists.txt'] = ' ',
	},
	pattern_matches = {
		['.*jquery.*.js$'] = ' ',
		['.*angular.*.js$'] = ' ',
		['.*backbone.*.js$'] = ' ',
		['.*require.*.js$'] = ' ',
		['.*materialize.*.js$'] = ' ',
		['.*materialize.*.css$'] = ' ',
		['.*mootools.*.js$'] = ' ',
		['Vagrantfile$'] = ' ',
	},
	}

	local function get_results_object(file_name)
		local abs_path = filepath.Join(dir, file_name)
		local icon = ""
		if is_dir(abs_path) then
			icon = devicons.extensions["folder"] or " "
		else
			if devicons.exact_matches[file_name] then
				icon = devicons.exact_matches[file_name]
			else
				local ext = file_name:match("^.+%.(.+)$")
				if ext and devicons.extensions[ext] then
					icon = devicons.extensions[ext]
				else
					icon = devicons.extensions["default"] or " "
				end
			end
		end
		return new_listobj(abs_path, icon, ownership, indent_n)
	end

keep in mind that there is no toggle to disable and enable the icons so u will have to add that yourself basically it just adds the font icons before the file name which makes them render. and yes it will only work if the user has the font enabled in his terminal which most people already do.
